### PR TITLE
Add new `status` command

### PR DESCRIPTION
**Pull request by DatCorno**

### CODE REVIEW

File: src/orion/core/cli/status.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+:mod:`orion.core.cli.status` -- Module to status experiments
+============================================================
+
+.. module:: status
+   :platform: Unix
+   :synopsis: List experiments in terminal
```
Comment (bouthilx) [inline src/orion/core/cli/status.py]: Copy paste? ;)

File: src/orion/core/cli/status.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    cli.get_basic_args_group(status_parser)
+
+    status_parser.add_argument(
+        '-a', '--all', action="store_true",
+        help="Show all trials line by line. Otherwise, they are all aggregated by status")
+
+    status_parser.add_argument(
+        '-r', '--recursive', action="store_true",
+        help="Divide trials per experiments hierarchically. Otherwise they are all aggregated in "
+             "parent experiment")
```
Comment (bouthilx) [inline src/orion/core/cli/status.py]: in root experiment?

Comment (DatCorno) [reply src/orion/core/cli/status.py]: Idem.

File: tests/functional/commands/conftest.py [hunk 3]
```diff
@@ -95,19 +99,194 @@ def database():
...
     """Clean the database and insert only experiments."""
-    database.experiments.drop()
     database.experiments.insert_many(exp_config[0])
-    database.trials.drop()
-    database.workers.drop()
-    database.resources.drop()
+
+
+def ensure_deterministic_id(name, db_instance, parent=None, root=None):
+    """Change the id of experiment `name` to `name`."""
```
Comment (bouthilx) [inline tests/functional/commands/conftest.py]:
```suggestion
    """Change the id of experiment to `name`."""
```

File: tests/functional/commands/conftest.py [hunk 4]
```diff
@@ -95,19 +99,194 @@ def database():
...
+    experiment = db_instance.read('experiments', {'name': name})[0]
+    db_instance.remove('experiments', {'_id': experiment['_id']})
+    experiment['_id'] = name
+    experiment['refers']['root_id'] = name
+
+    if parent is not None and root is None:
```
Comment (bouthilx) [inline tests/functional/commands/conftest.py]: That's a bit difficult to understand. You should document the args in the docstring. Also, should be `elif` I guess.

**Comment by DatCorno** [comment]
[add `orion status`](https://trello.com/c/NApMVgnT/26-add-orion-status)

**Comment by DatCorno** [comment]
[add `orion status`](https://trello.com/c/NApMVgnT/26-add-orion-status)

File: tests/functional/commands/test_list_command.py [hunk 5]
```diff
@@ -1,60 +1,12 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
-"""Perform a functional test of the list command."""
+"""perform a functional test of the list command."""
```
Comment (bouthilx) [inline tests/functional/commands/test_list_command.py]: Uppercase

File: tests/functional/commands/test_status_command.py [hunk 6]
```diff
@@ -0,0 +1,740 @@
...
+    assert captured == expected
+
+
+def test_experiment_wout_success_wout_ar(clean_db, single_without_success, capsys):
+    """Test status with only one experiment and no successful trial."""
+    orion.core.cli.main(['status'])
+
+    captured = capsys.readouterr().out
+
+    expected = """test_single_exp
```
Comment (bouthilx) [inline tests/functional/commands/test_status_command.py]:
```suggestion
    expected = """\
test_single_exp
```

File: tests/functional/commands/test_status_command.py [hunk 7]
```diff
@@ -0,0 +1,740 @@
...
+    assert captured == ""
+
+
+def test_experiment_without_trials_wout_ar(clean_db, one_experiment, capsys):
+    """Test status with only one experiment and no trials."""
+    orion.core.cli.main(['status'])
+
+    captured = capsys.readouterr().out
+
+    expected = """test_single_exp
```
Comment (bouthilx) [inline tests/functional/commands/test_status_command.py]:
```suggestion
    expected = """\
test_single_exp
```

**Pull request merged**